AnnService/inc/Core/Common/Dataset.h

```diff
@@ -73,6 +76,13 @@ namespace SPTAG
 
             inline const T* At(SizeType index) const
             {
+                if (index >= R())
```

Comment:
This check will be done thousands of times. We should remove it from critical path.